lib/trino-collect/src/main/java/io/trino/collect/cache/EvictableCache.java

```diff
@@ -200,14 +233,22 @@ public V putIfAbsent(K key, V value)
             @Override
             public boolean remove(Object key, Object value)
             {
-                // We could use delegate.compute(key, ..) to check existence and remove, but compute takes `K key` and we have `Object`
-                throw new UnsupportedOperationException();
+                @SuppressWarnings("SuspiciousMethodCalls") // Object passed to map as key K
+                Token<K> token = tokens.get(key);
+                if (token != null) {
```

Comment:
two threads calling at the same time this method will have the same result  right?  even if the dataCacheMap.remove gets called twice there's no problem that I see here.

Reply:
not sure i understand. can you describe the problem in more detail?

Reply:
I meant, if T1 and T2 all remove at the same time, the removal of the token from the cache will still happen - which it does.
I'm  still trying to figure out whether there could be a potential problem in a multithreading scenario when trying to update the two maps (tokens and cache).

Reply:
awesome. if you see a problematic scenario, let me know. thanks for looking into this